pkg/trace/stats/client_stats_aggregator.go

```diff
@@ -0,0 +1,283 @@
+package stats
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/watchdog"
+)
+
+const (
+	aggBucketSize          = uint64(1e9)
```

Comment:
I think it's misleading to call this variable aggBucketSize. It makes it look like changing it would change the size of the buckets, but flushOnTime actually assumes that buckets are always 1s.

Since it's used in a single place, I think a comment explaining why the timestamp is divided by 1e9 would be clearer. Or it should completely parametrize the bucket size in all the logic.